### PR TITLE
feat: Transfer Manager Metrics

### PR DESCRIPTION
**Pull request by danielbankhead**
🦕

### CODE REVIEW

**Comment by danielbankhead** [comment]
First commit is feature-only (no tests) to show it works w/o breaking anything. Working on tests next.

File: src/nodejs-common/util.ts [hunk 1]
```diff
@@ -537,7 +539,9 @@ export class Util {
           body: writeStream,
         },
       ],
-    } as {} as r.OptionsWithUri & GCCL_GCS_CMD;
+    } as {} as r.OptionsWithUri & {
```
Comment (ddelgrosso1) [inline src/nodejs-common/util.ts:542]: This is clever!

File: src/nodejs-common/util.ts [hunk 2]
```diff
@@ -36,6 +36,14 @@ import {getRuntimeTrackingString} from '../util';
 
 const packageJson = require('../../../package.json');
 
+/**
+ * A unique symbol for providing a `gccl-gcs-cmd` value
+ * for the `X-Goog-API-Client` header.
+ *
+ * E.g the `V` in `X-Goog-API-Client: gccl-gcs-cmd/V`
+ **/
+export const GCCL_GCS_CMD_KEY = Symbol('GCCL_GCS_CMD');
```
Comment (ddelgrosso1) [inline src/nodejs-common/util.ts:45]: Had to go look this up, didn't know `Symbol` was a thing.

**Review by danielduhh — COMMENTED** [review]
Approach LGTM

File: test/transfer-manager.ts [hunk 3]
```diff
@@ -413,14 +469,50 @@ describe('Transfer Manager', () => {
...
+              </InitiateMultipartUploadResult>`
+            ),
+            headers: {},
+          } as GaxiosResponse;
+        }
+      }
+
+      transferManager.bucket.storage.authClient = new GoogleAuth({
+        authClient: new TestAuthClient(),
+      });
```
Comment (danielbankhead) [inline test/transfer-manager.ts:513]: Makes it more of an integration test.

File: test/transfer-manager.ts [hunk 4]
```diff
@@ -255,10 +299,18 @@ describe('Transfer Manager', () => {
...
-        .stub(fs, 'createReadStream')
-        .returns(pThrough as unknown as fs.ReadStream);
+    before(async () => {
+      directory = await fsp.mkdtemp(
+        path.join(tmpdir(), 'tm-uploadFileInChunks-')
+      );
+
+      filePath = path.join(directory, 't.txt');
+
+      await fsp.writeFile(filePath, 'hello');
```
Comment (danielbankhead) [inline test/transfer-manager.ts:309]: Sharing a real, temp file to make it more robust. We can refactor the other tests if we want to use the real XML tool w/o exporting it.

File: test/transfer-manager.ts [hunk 5]
```diff
@@ -35,22 +35,28 @@ import * as stream from 'stream';
...
-    maxRetries: 3,
-    retryDelayMultiplier: 2,
-    totalTimeout: 600,
-    maxRetryDelay: 60,
-    retryableErrorFn: (err: ApiError) => {
-      return err.code === 500;
-    },
-    idempotencyStrategy: IdempotencyStrategy.RetryConditional,
-  };
+  const STORAGE = sinon.stub(
```
Comment (ddelgrosso1) [inline test/transfer-manager.ts:45]: Thanks, couldn't figure out the correct way to do this.

File: test/transfer-manager.ts (hunk 4 above)
Comment (ddelgrosso1) [reply test/transfer-manager.ts:309]: Yea this looks better than the hacky way I had it.

File: src/transfer-manager.ts [hunk 6]
```diff
@@ -183,28 +203,52 @@ class XMLMultiPartUploadHelper implements MultiPartUploadHelper {
...
+          headers[
+            key
+          ] = `${value} gccl-gcs-cmd/${GCCL_GCS_CMD_FEATURE.UPLOAD_SHARDED}`;
+        }
+        break;
+      }
+    }
+
+    // If the header isn't present, add it
+    if (!headerFound) {
```
Comment (danielduhh) [inline src/transfer-manager.ts:224]: We'd have to remember to update this line every time we add something new to headers. Is there a better way to do this?

Comment (danielbankhead) [reply src/transfer-manager.ts:224]: We can have a centralized place for this, however it would be cleaner to first complete the gaxios migration.

Comment (danielbankhead) [reply src/transfer-manager.ts:224]: This is also already matching the pattern in resumable uploads.

**Comment by danielbankhead** [comment]
PR's ready for review

File: src/transfer-manager.ts (hunk 6 above)
Comment (danielduhh) [reply src/transfer-manager.ts:224]: here's an example in [#2296](https://github.com/googleapis/nodejs-storage/blob/3ccc9bc9b87e5b537286ea8ce78ac39acb5a1290/src/nodejs-common/util.ts#L1023).

Yeah, thats fine. Please create a TODO & ticket to follow up

Comment (danielbankhead) [reply src/transfer-manager.ts:224]: https://github.com/googleapis/nodejs-storage/issues/2314

**Review by ddelgrosso1 — APPROVED** [review]

**Pull request merged**